xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -0,0 +1,148 @@
+package info.bitrich.xchangestream.kraken;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.kraken.dto.enums.KrakenEventType;
+import info.bitrich.xchangestream.kraken.dto.KrakenSubscriptionConfig;
+import info.bitrich.xchangestream.kraken.dto.KrakenSubscriptionMessage;
+import info.bitrich.xchangestream.kraken.dto.enums.KrakenSubscriptionName;
+import info.bitrich.xchangestream.kraken.dto.KrakenSubscriptionStatusMessage;
+import info.bitrich.xchangestream.kraken.dto.KrakenSystemStatus;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static info.bitrich.xchangestream.kraken.dto.enums.KrakenEventType.subscribe;
+
+/**
+ * @author makarid, pchertalev
+ */
+public class KrakenStreamingService extends JsonNettyStreamingService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KrakenStreamingService.class);
+    private static final String EVENT = "event";
+    private final Map<Integer, String> channels = new HashMap<>();
+    private ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+
+    private final Map<Integer, String> subscriptionRequestMap = new HashMap<>();
```

Comment:
Thread safety?

Reply:
Key value is random generated, map entry is used in waiting-for-subscription period. So I dont think that we need care about thrade safe here.

Reply:
https://stackoverflow.com/questions/2688629/is-a-hashmap-thread-safe-for-different-keys

Reply:
I have changed Map implemetation to ConcurrentHashMap

Reply:
:thumbsup:

Strictly then the following:

```
                    KrakenOrderBookStorage orderBook = ob.toKrakenOrderBook(orderBooks.get(channelName), depth);
                    orderBooks.put(channelName, orderBook);	                    orderBooks.put(channelName, orderBook);
```
Can be expressed as
```
orderBooks.compute(channelName, current -> ob.toKrakenOrderBook(current, depth));
```
But that's a much less likely race condition.

Reply:
it is imposible case when 2 different thread put the same key into this map. I used ConcurrentHashMap only because  map is used by different threads but withoput crossing by keys. So I don't see how compute method can improve this code.

Reply:
Fine